docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -5,21 +5,57 @@ lang: en
-| UnusedMethodRule | Path-based analysis | Pilot | Detects methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. |
+| Rule | Type | Catgegory | Status | Description |
+| -------- | ----------- | ----------- |----------- | ----------- |
+| ApexFlsViolationRule | Path-based analysis | Security | Generally Available (GA) | Detects Create, Read, Update, and Delete (CRUD) and Field-Level Security violations. |
+| ApexNullPointerExceptionRule | Path-based analysis | Error-Prone | GA | Identifies Apex operations that dereference null objects and throw NullPointerExceptions. |
+| AvoidDatabaseOperationInLoop | Path-based analysis | Performance | Pilot | Detects database operations in loops that degrade performance. |
+| AvoidMultipleMassSchemaLookups | Path-based analysis | Performance | GA | Detects scenarios where expensive schema lookups are made more than one time in a paths. |
+| RemoveUnusedMethod | Path-based analysis | Performance | Pilot | Detects methods contained in your code that aren’t invoked from any entry points that Graph Engine recognizes. |
+| UnimplementedTypeRule | Graph-based analysis | Performance | GA | Detects abstract classes and interfaces that are non-global and missing implementations or extensions. |
+| UseWithSharingOnDatabaseOperation | Path-based analysis | Security | Pilot | Detects database operations outside with-sharing-annotated classes. |
 
-## ApexFlsViolationRule
-ApexFlsViolationRule detects [Create, Read, Update, and Delete (CRUD) and Field-Level Security (FLS) violations](https://www.youtube.com/watch?v=1ZYjpjPTIn8). To run the path-based analysis, rules run `scanner:run:dfa`. Alternatively, run `scanner:run:dfa --category “Security”` to run only the ApexFlsViolationRule. 
+## Running Graph Engine GA Rules
+Run all Graph Engine rules against your code, or run a subset of rules by type or by category.
 
-Example: 
-```sfdx scanner:run:dfa --category "Security" --projectdir /project/dir --target /project/dir/target```
+* To run the path-based rules run: scanner:run:dfa --projectdir MyDirectory.
 
-### Definitions
+	**Example:**
+  
+	```sfdx scanner:run:dfa --projectdir /project/dir --target /project/dir/target1```
+
+* To run graph-based analysis rules run: scanner:run --engine sfge --projectdir MyDirectory.
+
+	**Example:**
+  
+	```sfdx scanner:run --engine sfge --projectdir /project/dir --target /project/dir/target1```
+
+* To run a category of rules, run: `scanner:run:dfa --category "Security"`
+  
+	**Example:**
+  
+	```sfdx scanner:run:dfa --category "Security" --projectdir /project/dir --target /project/dir/target```
+
+## Running Graph Engine Pilot Rules
+
+To run each Graph Engine pilot rule, include the ```--with-pilot``` flag in your request. 
+
+* To run all Graph Engine rules and all pilot rules, run:
+
+	```sfdx scanner:run:dfa --with-pilot --engine sfge --projectdir /project/dir --target /project/dir/target1```
```

Comment:
@jfeingold35 No, because I'm telling the user, To XYZ, run: [code]. Instead of: To XYZ do this. Example: [code]

Make sense?

Reply:
The other way to do this would be to separate the [code sample] from the intro sentence, but we don't have a lot more to say in the intro sentence that isn't repetitive. Or I can remove the separation and have the [code] appear right after the :